hedera-node/src/main/java/com/hedera/services/txns/token/TokenOpsValidator.java

```diff
@@ -83,15 +83,19 @@ public static ResponseCodeEnum validateTokenOpsWith(
 				return validity;
 			}
 			if (nftMetaDataCheck != null) {
-				for (Object bytes : metaDataList) {
-					validity = nftMetaDataCheck.apply( ((ByteString) bytes).toByteArray());
+				@SuppressWarnings("unchecked")
+				final List<ByteString> metadata = (List<ByteString>) metaDataList;
+				for (var bytes : metadata) {
+					validity = nftMetaDataCheck.apply(bytes.toByteArray());
```

Comment:
Thanks @tinker-michaelj for this PR. I saw this problem too when looking at PR #2137 a few days ago. I think a better solution is to also avoid the `@SuppressWarnings("unchecked")` and typecast to get the correct list. We could extract this `if then else` into two different helper functions. Extract the beginning into a common helper function and revise this function into having two different signatures which will consequently call the common helper function and then call the corresponding helper function. If that makes sense to you.

Reply:
Agree Quan, Will make changes in branch `02096-M-CodeCleanUp-ValidateMintBurnWipe` after I merge this PR

Reply:
@qnswirlds made changes here https://github.com/hashgraph/hedera-services/pull/2137/commits/904cc4a984b25dca69ba52f5f7b523669d35b65e